contract-examples/test/ExampleRewardDistributor.ts

```diff
@@ -0,0 +1,203 @@
+// (c) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import {
+  BigNumber,
+  Contract,
+  ContractFactory,
+} from "ethers"
+import { ethers } from "hardhat"
+import ts = require("typescript");
+
+// make sure this is always an admin for minter precompile
+const adminAddress: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+const REWARD_MANAGER_ADDRESS = "0x0200000000000000000000000000000000000004";
+const BLACKHOLE_ADDRESS = "0x0100000000000000000000000000000000000000";
+const rewardRate = ethers.utils.parseEther("0.00001")
+
+const ROLES = {
+  NONE: 0,
+  MINTER: 1,
+  ADMIN: 2
+};
+
+describe("ExampleRewardDistributor", function () {
+  this.timeout("30s")
+  let owner: SignerWithAddress
+  let contract: Contract
+  let signer1: SignerWithAddress
+  let signer2: SignerWithAddress
+  before(async function () {
+    owner = await ethers.getSigner(adminAddress);
+    signer1 = (await ethers.getSigners())[1]
+    signer2 = (await ethers.getSigners())[2]
+    const Contract: ContractFactory = await ethers.getContractFactory("ExampleRewardDistributor", { signer: owner })
+    contract = await Contract.deploy(rewardRate)
+    await contract.deployed()
+    const contractAddress: string = contract.address
+    console.log(`Contract deployed to: ${contractAddress}`)
+
+    // Send a transaction to mine a new block
+    const tx = await owner.sendTransaction({
+      to: signer1.address,
+      value: ethers.utils.parseEther("10")
+    })
+    await tx.wait()
+  });
+
+  it("should add contract deployer as owner", async function () {
+    const contractOwnerAddr: string = await contract.owner()
+    expect(owner.address).to.equal(contractOwnerAddr)
+  });
+
+  // this contract is not selected as the reward address yet, so should not be able to receive fees
+  it("contract should not be able to receive fees", async function () {
+    const rewardManager = await ethers.getContractAt("IRewardManager", REWARD_MANAGER_ADDRESS, owner);
+    let rewardAddress = await rewardManager.currentRewardAddress();
+    expect(rewardAddress).to.be.equal(BLACKHOLE_ADDRESS)
+
+    let balance = await ethers.provider.getBalance(contract.address)
+    expect(balance).to.be.equal(0)
+
+    // Send a transaction to mine a new block
+    const tx = await owner.sendTransaction({
+      to: signer1.address,
+      value: ethers.utils.parseEther("0.0001")
+    })
+    await tx.wait()
+
+    balance = await ethers.provider.getBalance(contract.address)
+    expect(balance).to.be.equal(0)
```

Comment:
Can we verify the updated balance of the blackhole address here as well?

Reply:
added